### PR TITLE
Implement ConstantOfShape

### PR DESCRIPTION
**Pull request by yuslepukhin**

### CODE REVIEW

**Comment by snnn** [comment]
Please also update onnxruntime/test/python/onnx_backend_test_series.py.

File: onnxruntime/core/providers/cpu/generator/constant_of_shape.cc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+using namespace ONNX_NAMESPACE;
+namespace onnxruntime {
+
+ONNX_CPU_OPERATOR_KERNEL(
+    ConstantOfShape,
+    9,
+    KernelDefBuilder()
+        .TypeConstraint("T1", DataTypeImpl::GetTensorType<int64_t>())
+        .TypeConstraint("T2", std::vector<MLDataType>{
+                                  DataTypeImpl::GetTensorType<MLFloat16>(),
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/generator/constant_of_shape.cc]: DataTypeImpl::AllFixedSizeTensorTypes() doesn't work?

File: onnxruntime/core/providers/cpu/generator/constant_of_shape.cc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  auto& input_shape = shape_tensor->Shape();
+
+  // If empty the output is a scalar, which means a single value
+  std::vector<int64_t> output_dims;
+  const auto input_size = input_shape.Size();
+  if (input_size > 0) {
+    auto span = gsl::make_span(shape_tensor->Data<int64_t>(), input_size);
+    output_dims.insert(output_dims.end(), span.cbegin(), span.cend());
+  } else if (input_size == 0) {
+    output_dims.push_back(1);  // scalar
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/generator/constant_of_shape.cc]: we represent scalars as a tensor with empty dimensions.

File: onnxruntime/core/providers/cpu/generator/constant_of_shape.cc (hunk 1 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/cpu/generator/constant_of_shape.cc]: That one includes BFloat16 now which is not in the spec. Not sure which one needs to be corrected.

**Pull request merged**